spring-cloud-gcp-samples/spring-cloud-gcp-data-datastore-sample/src/main/java/com/example/DatastoreRepositoryExample.java

```diff
@@ -49,16 +53,41 @@ public CommandLineRunner commandLineRunner() {
 			System.out.println("Remove all records from 'singers' kind");
 			this.singerRepository.deleteAll();
 
-			this.singerRepository.save(new Singer("singer1", "John", "Doe", ImmutableSet.of()));
```

Comment:
This example has become one big kitchen sink, which is very difficult to follow for me, let alone for someone new to Datastore. Is there any way this big method can be broken out into smaller sub-examples that it calls?

Reply:
The issue is with running these though. There has to be just one main method. 

Do you mean just moving some code into some private methods that get called in main like "demoTransactions" ?

Reply:
@meltsufin Also I gotta say it's kind of by-design. We specifically wanted to show all these advanced features in a sample, which I think might be contrary to the nature and purpose of samples in general.

Reply:
Moved some code into some smaller methods

Reply:
Thanks! Looks better!